ingestors/kafka/src/test/scala/hydra/kafka/algebras/KafkaClientAlgebraSpec.scala

```diff
@@ -67,68 +68,71 @@ class KafkaClientAlgebraSpec
       val (topic, (_, key), value) = topicAndKeyAndValue("topic1","key1","value1")
       (schemaRegistry.registerSchema(subject = s"$topic-key", key.getSchema) *>
         schemaRegistry.registerSchema(subject = s"$topic-value", value.getSchema) *>
-        kafkaClient.publishMessage((key, Some(value)), topic).map{ r =>
+        kafkaClient.publishMessage((key, Some(value), None), topic).map{ r =>
           r shouldBe PublishResponse(0, 0).asRight}).unsafeRunSync()
     }
 
     val (topic, (_, key), value) = topicAndKeyAndValue("topic1","key1","value1")
     "consume avro message from kafka" in {
       val records = kafkaClient.consumeMessages(topic,"newConsumerGroup").take(1).compile.toList.unsafeRunSync()
       records should have length 1
-      records.head shouldBe (key, Some(value))
+      records.head shouldBe (key, Some(value), Some(Headers.empty))
     }
 
     val (_, (_, key2), value2) = topicAndKeyAndValue("topic1","key2","value2")
     "publish avro record to existing topic and consume only that value in existing consumer group" in {
-      kafkaClient.publishMessage((key2, Some(value2)), topic).unsafeRunSync()
+      kafkaClient.publishMessage((key2, Some(value2), None), topic).unsafeRunSync()
       val records = kafkaClient.consumeMessages(topic, "newConsumerGroup6").take(2).compile.toList.unsafeRunSync()
-      records should contain allOf((key2, Some(value2)), (key, Some(value)))
+      records should contain allOf((key2, Some(value2), Some(Headers.empty)), (key, Some(value), Some(Headers.empty)))
```

Comment:
What is the difference between `Some(Headers.empty)` and `None`?

Reply:
Great question. Technically all records contain headers. So even if they are empty headers they technically exist. I can add additional logic that if the headers are empty to return None instead.

Reply:
My question is mostly just trying to understand if it is appropriate to have Option[Headers] or if we should just have Headers since there is an empty header? Thoughts?